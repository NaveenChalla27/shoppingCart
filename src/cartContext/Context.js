
import { createContext,useContext, useReducer } from "react";
import {faker} from "@faker-js/faker"
import React from 'react';
import CartReducer from "./Reducer";
import ProductReducer from "./productReducer";

export const Cart=createContext();
faker.seed(99)

const Context = ({children}) => {
const products=[...Array(20)].map(()=>({
    id:faker.datatype.uuid(),
    name:faker.commerce.product(),
    price:faker.commerce.price(),
    image:faker.image.fashion(),
    fastDelivery:faker.datatype.boolean(),
    inStock:faker.helpers.arrayElement([0,3,5,6,7]),
    rating:faker.helpers.arrayElement([1,2,3,4,5])

}))

const [state,dispatch]=useReducer(CartReducer,{products,cart:[]})
const [productState,productDispatch]=useReducer(ProductReducer,{
  byStock:false,
  byFastDelivery:false,
  byRating:0,
  searchQuery:""
})
  return (
     <Cart.Provider value={{state,dispatch,productState,productDispatch}}>
        {children},
     </Cart.Provider>
  )
}

export default Context

export const CartState=()=>{
    return useContext(Cart);
}



