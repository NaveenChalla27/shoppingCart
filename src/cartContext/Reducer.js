const cartReducer=(state,action)=>{
    switch( action.type){
        case "ADD_TO_CART":
            return {...state,cart:[...state.cart,{...action.payload,qty:1}]}
        case "REMOVE_FROM_CART":
             return {...state,cart:state.cart.filter(item=>item.id!==action.payload.id)}
        case  "CHANGE_CART_QTY":
            //   console.log(action.payload)
            //   return {
            //     ...state,
            //     cart: state.cart.filter((c) =>
            //       c.id === action.payload.id ? (c.qty = action.payload.qty) : c.qty
            //     ),
            //   };
              return {...state,cart:state.cart.filter(item=>{ return item.id===action.payload.id ? (item.qty=action.payload.qty):item.qty})}
              case  "CHANGE_CART_RATING":
                //return {...state,cart:state.cart.filter((item)=>item.id===action.payload.id?(item.rating===action.payload.rating):item.rating)}
                   return {
                ...state,
                cart: state.cart.filter((c) =>
                  c.id === action.payload.id ? (c.rating = action.payload.rating) : c.rating
                ),
              };
        default:
            return state;
    }
 }
export default cartReducer
