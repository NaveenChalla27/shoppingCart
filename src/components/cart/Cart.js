import React, { useEffect,useState } from 'react';
import { CartState } from "../../cartContext/Context";
import { Button, Col,Row, ListGroup,Form ,Image} from 'react-bootstrap';
import Rating from "../rating/Rating";
import { AiFillDelete } from 'react-icons/ai';
import "./cart.css"

const Cart = () => {
   const {state:{cart},dispatch}=CartState();
   const [total,setTotal]=useState(0);

  useEffect(()=>{
    let data=cart.reduce((acc,curr)=>acc+Number(curr.price)*curr.qty,0)
    setTotal(data)
   },[cart])

const changeQuantity=(e,prod)=>{
  dispatch({ type: "CHANGE_CART_QTY",payload:{
    id:prod.id,
    qty:e.target.value}
  })

}
  return (
    <div className='homePage'>
    <div className='productsContainer'>
      <ListGroup>
        {
          cart.map(prod=>(
            <ListGroup.Item key={prod.id}>
             <Row>
               <Col md={2}>
                 <Image src={prod.image} alt={prod.image} fluid rounded/>
               </Col>
                <Col md={2} >
                  <span>{prod.name}</span>
                </Col>
                <Col md={2} >${prod.price}</Col>
                <Col md={2} >
                <Rating rating={prod.rating} onClick={(value)=>{dispatch({type:"CHANGE_CART_RATING",payload:{id:prod.id,rating:value}})}}></Rating>
                </Col>
                <Col md={2}>
                <Form.Control as="select" value={prod.qty} onChange={(e)=>changeQuantity(e,prod)}>
                {[...Array(prod.inStock).keys()].map((x)=>(
                  <option key={x+1}>{x+1}</option>
                ))}
                </Form.Control>
                </Col>
                <Col md={2}>
                  <Button type="button" variant="light" onClick={()=>dispatch({type:"REMOVE_FROM_CART",payload:prod})}>
                   <AiFillDelete fontSize={"20px"}/>
                  </Button>
                </Col>
             </Row>
            </ListGroup.Item>
            ))
        }
      </ListGroup>
    </div>
    <div className='filter checkout'>
      <span className='title'>Subtotal ({cart.length}) items</span>
      <span style={{fontWeight:600,fontSize:18}}>Total:${total}</span>
       <Button type="button" disabled={cart.length===0}>Proceed to Checkout</Button>
    </div>
    </div>
  );
};

export default Cart