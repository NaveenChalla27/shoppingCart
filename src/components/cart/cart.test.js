/* eslint-disable testing-library/no-debugging-utils */
import Cart from "./Cart";
import { Cart as CartContext } from "../../cartContext/Context";
import { render, screen } from "@testing-library/react";
describe("cart", () => {
 test("cart render",()=>{
  const dispatch = jest.fn();
  expect(render(
    <CartContext.Provider
      value={{
        dispatch,
        state: {
          cart: [
            {
              fastDelivery: true,
              id: "06f5602e-30b3-45c9-af27-08344bcb9606",
              image: "https://loremflickr.com/640/480/fashion",
              inStock: 6,
              name: "Shirt",
              price: "675.00",
              rating: 1,
              qty: 1,
            },
          ],
        },
      }}
    >
      <Cart />
    </CartContext.Provider>
  )).toMatchSnapshot();
})


});
