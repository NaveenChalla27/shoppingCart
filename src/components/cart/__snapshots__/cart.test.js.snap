// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cart cart render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="homePage"
      >
        <div
          class="productsContainer"
        >
          <div
            class="list-group"
          >
            <div
              class="list-group-item"
            >
              <div
                class="row"
              >
                <div
                  class="col-md-2"
                >
                  <img
                    alt="https://loremflickr.com/640/480/fashion"
                    class="img-fluid rounded"
                    src="https://loremflickr.com/640/480/fashion"
                  />
                </div>
                <div
                  class="col-md-2"
                >
                  <span>
                    Shirt
                  </span>
                </div>
                <div
                  class="col-md-2"
                >
                  $
                  675.00
                </div>
                <div
                  class="col-md-2"
                >
                  <div>
                    <span
                      data-testid="rating"
                    >
                      <svg
                        fill="currentColor"
                        font-size="15px"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 1024 1024"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                    <span
                      data-testid="rating"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 1024 1024"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                        />
                      </svg>
                    </span>
                    <span
                      data-testid="rating"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 1024 1024"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                        />
                      </svg>
                    </span>
                    <span
                      data-testid="rating"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 1024 1024"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                        />
                      </svg>
                    </span>
                    <span
                      data-testid="rating"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 1024 1024"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
                <div
                  class="col-md-2"
                >
                  <select
                    class="form-control"
                  >
                    <option>
                      1
                    </option>
                    <option>
                      2
                    </option>
                    <option>
                      3
                    </option>
                    <option>
                      4
                    </option>
                    <option>
                      5
                    </option>
                    <option>
                      6
                    </option>
                  </select>
                </div>
                <div
                  class="col-md-2"
                >
                  <button
                    class="btn btn-light"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      font-size="20px"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="filter checkout"
        >
          <span
            class="title"
          >
            Subtotal (
            1
            ) items
          </span>
          <span
            style="font-weight: 600; font-size: 18px;"
          >
            Total:$
            675
          </span>
          <button
            class="btn btn-primary"
            type="button"
          >
            Proceed to Checkout
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="homePage"
    >
      <div
        class="productsContainer"
      >
        <div
          class="list-group"
        >
          <div
            class="list-group-item"
          >
            <div
              class="row"
            >
              <div
                class="col-md-2"
              >
                <img
                  alt="https://loremflickr.com/640/480/fashion"
                  class="img-fluid rounded"
                  src="https://loremflickr.com/640/480/fashion"
                />
              </div>
              <div
                class="col-md-2"
              >
                <span>
                  Shirt
                </span>
              </div>
              <div
                class="col-md-2"
              >
                $
                675.00
              </div>
              <div
                class="col-md-2"
              >
                <div>
                  <span
                    data-testid="rating"
                  >
                    <svg
                      fill="currentColor"
                      font-size="15px"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                      />
                    </svg>
                  </span>
                  <span
                    data-testid="rating"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                      />
                    </svg>
                  </span>
                  <span
                    data-testid="rating"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                      />
                    </svg>
                  </span>
                  <span
                    data-testid="rating"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                      />
                    </svg>
                  </span>
                  <span
                    data-testid="rating"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <div
                class="col-md-2"
              >
                <select
                  class="form-control"
                >
                  <option>
                    1
                  </option>
                  <option>
                    2
                  </option>
                  <option>
                    3
                  </option>
                  <option>
                    4
                  </option>
                  <option>
                    5
                  </option>
                  <option>
                    6
                  </option>
                </select>
              </div>
              <div
                class="col-md-2"
              >
                <button
                  class="btn btn-light"
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    font-size="20px"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="filter checkout"
      >
        <span
          class="title"
        >
          Subtotal (
          1
          ) items
        </span>
        <span
          style="font-weight: 600; font-size: 18px;"
        >
          Total:$
          675
        </span>
        <button
          class="btn btn-primary"
          type="button"
        >
          Proceed to Checkout
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
